packages/babel-types/src/clone/cloneNode.js

```diff
@@ -34,10 +34,7 @@ export default function cloneNode<T: Object>(node: T, deep: boolean = true): T {
   const { type } = node;
   const newNode = (({ type }: any): T);
 
-  // Special-case identifiers since they are the most cloned nodes.
-  if (type === "Identifier") {
```

Comment:
Same here: I will maybe simply create a bug report that this function does not clone all fields of an Identifier node correctly and we discuss there what to do about it. This really just came up because the test case that is testing this function is testing the optimised case and fails if `t.identifier` takes additional (nullable) arguments.